frontend/src/components/transactions/TransactionDetails.tsx

```diff
@@ -71,31 +68,62 @@ export default class extends React.Component<Props, State> {
     return gasBurntByTx.add(gasBurntByReceipts);
   }
 
+  updateBlock = async () => {
+    const block = await new BlocksApi().getBlockInfo(
+      this.props.transaction.blockHash
+    );
+    this.setState({ block });
+  };
+
   updateComputedValues = () => {
     const deposit = this.collectDeposit(this.props.transaction.actions);
     const gasUsed = this.collectGasUsed(this.props.transaction);
-    const gasPrice = new BN(this.props.transaction.gasPrice);
-    const transactionFee = gasUsed.mul(gasPrice);
     let gasAttached = this.collectGasAttached(this.props.transaction.actions);
     if (gasAttached === null) {
       gasAttached = gasUsed;
     }
-    this.setState({ deposit, transactionFee, gasUsed, gasAttached });
+    const stateUpdate: any = { deposit, gasUsed, gasAttached };
+    if (this.state.block) {
+      stateUpdate.transactionFee = gasUsed.mul(
```

Comment:
I am not very sure about this: stateUpdate not have transactionFee included, is this legal ?